crates/testing/src/predicates/event.rs

```diff
@@ -25,6 +26,89 @@ impl<TYPES: NodeType> std::fmt::Debug for EventPredicate<TYPES> {
     }
 }
 
+#[allow(clippy::type_complexity)]
+pub struct TestPredicate<INPUT> {
```

Comment:
Would `AllPredicates` be a better name?

Reply:
It's a more general type than just the `all` and `all_predicates`; it should accommodate basically any predicate.